data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/MultipartCatalogDescriptionRequestSender.java

```diff
@@ -55,10 +55,41 @@ public MultipartCatalogDescriptionRequestSender(@NotNull String connectorId,
                                                     @NotNull ObjectMapper objectMapper,
                                                     @NotNull Monitor monitor,
                                                     @NotNull IdentityService identityService,
-                                                    @NotNull TransformerRegistry transformerRegistry) {
+                                                    @NotNull IdsTransformerRegistry transformerRegistry) {
         super(connectorId, httpClient, objectMapper, monitor, identityService, transformerRegistry);
     }
 
+    private static void createOfferResourcesFromProperties(ResourceCatalog catalog, ObjectMapper mapper) {
+        if (catalog.getProperties() != null) {
+            for (Map.Entry<String, Object> entry : catalog.getProperties().entrySet()) {
+                if ("ids:offeredResource".equals(entry.getKey())) {
+                    JsonNode node = mapper.convertValue(entry.getValue(), JsonNode.class);
+                    List<Resource> offeredResources = new LinkedList<>();
+                    for (JsonNode objNode : node.get("objectList")) {
+                        Map<String, Object> resource = new HashMap<>();
+                        resource.put("@type", "ids:Resource");
+                        resource.putAll(mapper.convertValue(objNode, Map.class));
+                        offeredResources.add(mapper.convertValue(resource, Resource.class));
+                    }
+                    catalog.setOfferedResource(offeredResources);
+                }
+            }
+        }
+    }
+
+    private static boolean catalogDoesNotContainAnyOfferResource(ResourceCatalog catalog) {
```

Comment:
This can be moved as instance method in `ResourceCatalog` class

Reply:
this pops up as modified because of a rebase, so while I agree with it, I'd prefer to do this in another PR. This holds true for all the other IDS-related classes as well.